### PR TITLE
Make SSL truststore and keystore configuration consistent across connectors

### LINKED ISSUE #17103: Make SSL truststore/keystore configuration consistent across connectors
In different connectors we are using different names for the same configuration properties. It'd be nice to make them consistent by introducing a single configuration class (`SslTrustConfig`) in the `plugin-toolkit` that could be applied to multiple connectors using `ConfigBinder`.`public <T> void bindConfig(Class<T> configClass, String prefix)` method (this allows us to add a unique prefix like `mongodb.tls`)

Since names right now are inconsistent we should settle on a single, consistent name and add rest of those as `@LegacyConfig`

```
plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java:    @Config("cassandra.tls.truststore-path")
plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java:    @Config("cassandra.tls.truststore-password")
plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/security/KafkaSslConfig.java:    @Config("kafka.ssl.truststore.location")
plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/security/KafkaSslConfig.java:    @Config("kafka.ssl.truststore.password")
plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/security/KafkaSslConfig.java:    @Config("kafka.ssl.truststore.type")
plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSslConfig.java:    @Config("mongodb.tls.truststore-path")
plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSslConfig.java:    @Config("mongodb.tls.truststore-password")
plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConfig.java:    @Config("elasticsearch.tls.truststore-path")
plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConfig.java:    @Config("elasticsearch.tls.truststore-password")
plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcServerQueryClientTlsConfig.java:    @Config("pinot.grpc.tls.truststore-type")
plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcServerQueryClientTlsConfig.java:    @Config("pinot.grpc.tls.truststore-path")
plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcServerQueryClientTlsConfig.java:    @Config("pinot.grpc.tls.truststore-password")
core/trino-main/src/main/java/io/trino/server/InternalCommunicationConfig.java:    @Config("internal-communication.https.truststore.path")
core/trino-main/src/main/java/io/trino/server/InternalCommunicationConfig.java:    @Config("internal-communication.https.truststore.key")
lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/LdapClientConfig.java:    @Config("ldap.ssl.truststore.path")
lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/LdapClientConfig.java:    @Config("ldap.ssl.truststore.password")```

### PR DESCRIPTION
**Pull request by abusk**
## Description
This PR creates a separate SSL config to make ssl/tls  configuration consistent across connectors. 

## Additional context and related issues
Fixes #17103 

## Release notes

(x ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`17103`)
```

### CODE REVIEW

**Comment by abusk** [comment]
Will update the ssl config for the rest of the connectors(Listed in the issue [17103](https://github.com/trinodb/trino/issues/17103)) after this PR revision reviewed.

**Review by hashhar — COMMENTED** [review]
just skimmed but looks very nice.

do you plan to change other connectors to use this as well?

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraClientConfig.java [hunk 1]
```diff
@@ -99,10 +88,6 @@ public void testExplicitPropertyMappings()
                 .put("cassandra.speculative-execution.delay", "101s")
                 .put("cassandra.protocol-version", "V3")
                 .put("cassandra.tls.enabled", "true")
-                .put("cassandra.tls.keystore-path", keystoreFile.toString())
-                .put("cassandra.tls.keystore-password", "keystore-password")
-                .put("cassandra.tls.truststore-path", truststoreFile.toString())
-                .put("cassandra.tls.truststore-password", "truststore-password")
```
Comment (hashhar) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraClientConfig.java:105]: can you add a test which verifies that if Cassandra connector is started with the old names it still works?

Comment (abusk) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraClientConfig.java:105]: Yes I have tested with debugger in the `TestCassandraConnector` and it works for the legacy configs too. But adding a test case will be tricky here, as the connectors bootstrap with valid certificate only.  I don't see any certificates or test cases enabling TLS here.

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        this.truststorePassword = truststorePassword;
+        return this;
+    }
+
+    public Optional<TruststoreType> getTruststoreType()
+    {
+        return Optional.ofNullable(truststoreType);
+    }
+
+    @Config("truststore-type")
```
Comment (ebyhr) [inline lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java:92]: This property is unused from Cassandra connector. Binding such property doesn't make sense.

**Review by ebyhr — CHANGES_REQUESTED** [review]

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraClientConfig.java (hunk 1 above)
Comment (wendigo) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraClientConfig.java:105]: https://github.com/trinodb/trino/pull/20430

**Comment by ebyhr** [comment]
Please remove unrelated commits.

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java (hunk 2 above)
Comment (ebyhr) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java:108]: Reminder. The connector still accepts unsupported config properties.

**Comment by abusk** [comment]
Made common ssl config for all the connectors mentioned in the issue description([17103](https://github.com/trinodb/trino/issues/17103)), except for the `Ldap` and the `Internal communication` in the server

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java (hunk 2 above)
Comment (abusk) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java:108]: Yea, taken into consideration all the legacy properties.

**Comment by abusk** [comment]
@hashhar @ebyhr please review.

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java (hunk 2 above)
Comment (ebyhr) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java:108]: It's not fixed. The connector shouldn't allow `*-type` config properties.

**Review by ebyhr — CHANGES_REQUESTED** [review]
Please take a look at https://github.com/trinodb/trino/pull/20409#discussion_r1460099156

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java (hunk 2 above)
Comment (abusk) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ssl/SslTrustConfig.java]: Made the change to remove unsupported properties.

**Comment by abusk** [comment]
@ebyhr please review again

**Comment by mosabua** [comment]
We should continue on this in my opinion. What is needed as next step @ebyhr or @wendigo ?

**Pull request closed without merging**